### ISSUE TITLE
[libdart] NLOPT::header and IPOPT::header are not found

### ISSUE THREAD

**Issue by Bidski** [acxz/pkgbuilds#71]
Trying to build libdart with makepkg on Arch Linux results in the following output

```
==> Downloading 'libdart.tar.gz' to '/home/bidski/aur_builds'
==> Extracting '/home/bidski/aur_builds/libdart.tar.gz'
==> Making package: libdart 6.9.4-1 (Mon 07 Sep 2020 09:16:25)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Retrieving sources...
  -> Downloading libdart-6.9.4.tar.gz...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   120  100   120    0     0    359      0 --:--:-- --:--:-- --:--:--   359
100 14.1M    0 14.1M    0     0  4292k      0 --:--:--  0:00:03 --:--:-- 5559k
==> Validating source files with sha256sums...
    libdart-6.9.4.tar.gz ... Passed
==> Extracting sources...
  -> Extracting libdart-6.9.4.tar.gz with bsdtar
==> Starting build()...
-- The C compiler identification is GNU 10.2.0
-- The CXX compiler identification is GNU 10.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- 
-- ============================================
--                 DART 6.9.4
-- ============================================
-- Found fcl: /usr/include (found suitable version "0.6.1", minimum required is "0.3.2") 
-- Found assimp: /usr/include (found version "5.0.0") 
-- Performing Test ASSIMP_AISCENE_CTOR_DTOR_DEFINED
-- Performing Test ASSIMP_AISCENE_CTOR_DTOR_DEFINED - Success
-- Performing Test ASSIMP_AIMATERIAL_CTOR_DTOR_DEFINED
-- Performing Test ASSIMP_AIMATERIAL_CTOR_DTOR_DEFINED - Success
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE  
-- Required Boost libraries: serialization
-- Found Boost: /usr/lib64/cmake/Boost-1.72.0/BoostConfig.cmake (found suitable version "1.72.0", minimum required is "1.60.0") found components: serialization 
-- Detected Boost version: 1.72.0
-- Boost include dirs: /usr/include
-- Found NLOPT: /usr/include (found version "2.6.2") 
-- Found IPOPT: /usr/include/coin-or (found version "3.13.2") 
-- Performing Test PAGMO_BUILT_WITH_NLOPT
CMake Error at /media/RAID/aur_builds/libdart/src/dart-6.9.4/build/CMakeFiles/CMakeTmp/CMakeLists.txt:18 (add_executable):
  Target "cmTC_c567a" links to target "NLOPT::header" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?


CMake Error at /media/RAID/aur_builds/libdart/src/dart-6.9.4/build/CMakeFiles/CMakeTmp/CMakeLists.txt:18 (add_executable):
  Target "cmTC_c567a" links to target "IPOPT::header" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?


CMake Error at /usr/share/cmake-3.18/Modules/CheckCXXSourceCompiles.cmake:110 (try_compile):
  Failed to generate test project build system.
Call Stack (most recent call first):
  cmake/DARTFindpagmo.cmake:22 (check_cxx_source_compiles)
  cmake/DARTMacros.cmake:81 (include)
  dart/optimizer/pagmo/CMakeLists.txt:2 (dart_find_package)


-- Configuring incomplete, errors occurred!
See also "/media/RAID/aur_builds/libdart/src/dart-6.9.4/build/CMakeFiles/CMakeOutput.log".
See also "/media/RAID/aur_builds/libdart/src/dart-6.9.4/build/CMakeFiles/CMakeError.log".
==> ERROR: A failure occurred in build().
    Aborting...
```

Installed versions
assimp: 5.0.1-2
boost: 1.72.0-2
eigen: 3.3.7-7
fcl: 0.6.1-1
libccd: 2.1-1
cmake: 3.18.2-1
bullet: 2.89-2
coin-or-ipopt: 3.13.2-1
flann: 1.9.1-7
freeglut: 3.2.1-2
libxi: 1.7.10-3
libxmu: 1.1.3-2
nlopt: 2.6.2-1
octomap: 1.9.5-1
ode: 0.16.1-1
openscenegraph: 3.6.5-1
pagmo: r4289.30257c05-1
tinyxml2: 7.1.0-1
urdfdom: 1.0.4-4

**Comment by acxz** [comment]
hmm thx for the reporting the issue @Bidski but I can't tell whats wrong from a glance. Could you make an upstream issue as well? I feel like they may know what is going on here.

**Comment by Bidski** [comment]
Looks like it is upstream in pagmo.
[Dartsim issue](https://github.com/dartsim/dart/issues/1473)
[Pagmo issue](https://github.com/esa/pagmo2/issues/445)

**Comment by bluescarni** [comment]
I think that the issue is that pagmo and dartsim both have ``FindNLOPT.cmake`` and ``FindIPOPT.cmake`` files for locating the NLopt and IPOPT dependencies.

In the upcoming version of pagmo, 2.16, we have removed the ``FindNLOPT.cmake`` files and we are now relying on the CMake package files installed by NLopt itself. As a consequence, pagmo now requires NLopt 2.6 or later.

Regarding IPOPT, we have renamed ``FindIPOPT.cmake`` to ``FindIpagmo_POPT.cmake``, and in theory that *should* take care of the name clashing. It would be great if someone could test this theory though :)

The relevant PR in pagmo is https://github.com/esa/pagmo2/pull/451 which should be merged shorty.

**Comment by acxz** [comment]
@Bidski pagmo has been updated to 2.16.0 can you confirm if this error still persists?

**Comment by Bidski** [comment]
@acxz I see the same error

**Comment by Bidski** [comment]
Unrelated to this particular issue, but still related to pagmo. https://github.com/esa/pagmo2/issues/456

**Comment by acxz** [comment]
@Bidski can you post a separate issue about the C++17 issue? Thx.

**Comment by acxz** [comment]
patched with ff25c45

**Comment by acxz** [comment]
Closing this access as lost access to libdart
